### PR TITLE
Logschema native parsers

### PR DESCRIPTION
**Pull request by alxarch**
## Background

This PR includes the foundations for Managed Schemas

## Changes

- Adds `mage build:schemas` to export native log types as schemas in `out/schemas`
- Adds `NativeParser` configuration in `logschema.Parser`
- Adds `logschema.InferGoTypeValueSchema` to infer the schema spec from go structs
- Adds `structfields` package to include common `reflect` tasks when working with schema structs (used in `glueschema` also)

## Testing

- mage test:go
- ran mage build:schemas and browsed through the produced schema catalog

### CODE REVIEW

File: internal/log_analysis/awsglue/glueschema/glueschema.go [hunk 1]
```diff
@@ -231,32 +208,14 @@ func fieldComment(field *reflect.StructField) string {
...
 	if err != nil {
 		return "", err
 	}
-	switch tag, err := tags.Get("json"); {
-	case err != nil:
-		// Field has no JSON struct tag, it uses the same name in JSON
-		return ColumnName(field.Name), nil
-	case tag.Name == "-":
-		// Field is explicitly omitted for JSON
+	if name == "" {
```
Comment (kostaspap) [inline internal/log_analysis/awsglue/glueschema/glueschema.go:215]: Can you share some context about the changes in `fieldColumnName` function? Why is it required? What does it do? It is a bit hard to understand the logic.

File: internal/log_analysis/log_processor/logschema/infer.go [hunk 2]
```diff
@@ -165,3 +170,90 @@ func (v *ValueSchema) NonEmpty() *ValueSchema {
...
+	}
+	// Handle 'special types'
+	if valueType, ok := inverseMappings[typ]; ok {
+		return &ValueSchema{
+			Type: valueType,
+		}, nil
+	}
+	switch typ.Kind() {
+	case reflect.Struct:
+		if typ.ConvertibleTo(reflect.TypeOf(time.Time{})) {
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/logschema/infer.go:187]: Why is the `ConvertibleTo` required here?

File: internal/log_analysis/log_processor/logschema/reflect.go [hunk 3]
```diff
@@ -46,6 +46,14 @@ var (
 		TypeBoolean:   reflect.TypeOf(pantherlog.Bool{}),
 		TypeTimestamp: reflect.TypeOf(pantherlog.Time{}),
 	}
+	// used in InferGoTypeValueSchema
+	inverseMappings = func() map[reflect.Type]ValueType {
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/logschema/reflect.go:50]: Can you add a comment describing what this function does and why it is needed?

File: internal/log_analysis/log_processor/registry/export.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/panther-labs/panther/internal/log_analysis/log_processor/logschema"
+)
+
+func ExportSchemas() (map[string]*logschema.Schema, error) {
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/registry/export.go:30]: Please add some documentation on what this function does.

File: internal/log_analysis/log_processor/registry/internal/export.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	if pos := strings.IndexByte(name, '.'); 0 <= pos && pos < len(name) {
+		return name[:pos], name[pos+1:]
+	}
+	return "", name
+}
+
+func OutputFileName(name string) string {
+	group, name := splitName(name)
+	return path.Join(strcase.ToSnake(group), strcase.ToSnake(name)+".yml")
+}
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/registry/internal/export.go:38]: Are these functions used?

File: pkg/x/structfields/structfields.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	"strings"
+
+	"github.com/fatih/structtag"
+)
+
+func Flatten(typ reflect.Type) []reflect.StructField {
+	return appendStructFields(nil, typ)
+}
+
+func FieldNameJSON(field reflect.StructField) (string, error) {
```
Comment (kostaspap) [inline pkg/x/structfields/structfields.go:33]: Please add documentation for these methods. The code is not trivial for someone reading it for first time, any comments here would help

**Review by kostaspap — COMMENTED** [review]
Good stuff

File: internal/log_analysis/log_processor/logschema/infer.go (hunk 2 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/logschema/infer.go:187]: For cases such as
```golang
type FooBarTimestamp time.Time
```

File: internal/log_analysis/log_processor/logschema/reflect.go (hunk 3 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/logschema/reflect.go:50]: The function does not do anything. It is called immediately to return the `map[reflect.Type]ValueType`

File: internal/log_analysis/log_processor/registry/export.go (hunk 4 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/registry/export.go:33]: Added

File: internal/log_analysis/log_processor/registry/internal/export.go (hunk 5 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/registry/internal/export.go]: Removed

File: pkg/x/structfields/structfields.go (hunk 6 above)
Comment (alxarch) [reply pkg/x/structfields/structfields.go]: Added

File: internal/log_analysis/awsglue/glueschema/glueschema.go (hunk 1 above)
Comment (alxarch) [reply internal/log_analysis/awsglue/glueschema/glueschema.go:215]: There are no changes in the logic. The requirement to have the JSON field name was needed in `logschema.InferTypeValueSchema` so I 'lifted' the logic to a separate function in `structfields` package

File: internal/log_analysis/log_processor/logschema/infer.go (hunk 2 above)
Comment (kostaspap) [reply internal/log_analysis/log_processor/logschema/infer.go:187]: Ah makes sense. Could you add a comment in the code about this?

**Review by kostaspap — APPROVED** [review]
There is only a minor comment open from my side (about adding some comment in the code), LGTM otherwise!

File: internal/log_analysis/log_processor/logschema/infer.go (hunk 2 above)
Comment (alxarch) [reply internal/log_analysis/log_processor/logschema/infer.go:188]: Added

**Pull request merged**